lib/include/openamp/virtio.h

```diff
@@ -499,6 +511,49 @@ static inline int virtio_reset_device(struct virtio_device *vdev)
 	return 0;
 }
 
+/**
+ * @brief Allocate buffer from the virtio device
+ *
+ * @param vdev	Pointer to virtio device structure.
+ * @param buf	Pointer to the allocated buffer.
+ * @param size	Allocated buffer size.
+ * @param align	Allocated buffer alignment.
+ *
+ * @return 0 on success, otherwise error code.
+ */
+static inline int virtio_alloc_buf(struct virtio_device *vdev, void **buf,
+				   size_t size, size_t align)
+{
+	if (!vdev || !buf)
+		return -EINVAL;
+
+	if (!vdev->mmops || !vdev->mmops->alloc)
+		return -ENXIO;
+
+	*buf = vdev->mmops->alloc(vdev, size, align);
```

Comment:
If *buf == NULL, then this should return -ENOMEM.

Reply:
Thanks, Done